packages/react/src/components/input/input.tsx

```diff
@@ -1,21 +1,48 @@
 import { c, classy, InputProps as BaseProps, m } from '@onfido/castor';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { FieldLabelWrapper } from '../../internal';
 import { withRef } from '../../utils';
 
+const idPrefix = 'castor_input';
+let idCount = 0;
+
 export const Input = withRef(
   (
-    { type = 'text', invalid, className, ...restProps }: InputProps,
+    {
+      id: externalId,
+      type = 'text',
+      invalid,
+      children,
+      className,
+      ...restProps
+    }: InputProps,
     ref: InputProps['ref']
-  ): JSX.Element => (
-    <input
-      {...restProps}
-      ref={ref}
-      type={type}
-      className={classy(c('input'), m({ invalid }), className)}
-    />
-  )
+  ): JSX.Element => {
+    const [id, setId] = useState<string | undefined>(externalId);
+
+    useEffect(() => {
+      if (externalId || children)
+        setId(externalId || `${idPrefix}_${++idCount}`);
+    }, [externalId]);
```

Comment:
There's a few bugs in here, and there's also an extra re-render when `externalId` changes.
We can get rid of the effect and simplify it a bit by using a state initialiser function.

```suggestion
    const [autoId, setAutoId] = useState(() => children && `${idPrefix}_${++idCount}`);
    const id = externalId || autoId;
```

Reply:
Interesting, but this won't catch the `children` change would it?

Reply:
It wouldn't, but that's better because the id is constant regardless of how many times `children` changes.
There's no harm in applying the id if the `children` is for some reason later removed, but if you really wanted to you could with
```ts
const id = externalId || (children ? autoId : undefined);
```